frog/imports/ui/GraphEditor/store/uiStore.js

```diff
@@ -1,14 +1,23 @@
 // @flow
-import { action, observable, computed } from 'mobx';
+import { action, observable, computed, autorun } from 'mobx';
 import { between, timeToPx, pxToTime } from '../utils';
 import { store } from './index';
 import type { Elem } from './store';
 import Operator from './operator';
 
 export default class uiStore {
+  constructor() {
+    autorun(() => {
+      if (this.selected) {
+        this.setStickySelected(this.selected);
+      }
+    });
```

Comment:
Right now, I am quite liberal about using cancelAllSelections or what it's called, and the result is for example that the sidebar (which relies on the currently selected Activity/Operator) would disappear - for example by double-clicking to rename an activity, and pressing enter to complete the rename (not so common in real life perhaps, but common when I want to demonstrate how the two title fields are linked ;))...

autorun is a mobx functionality that tracks all the dependencies (ie. all the mobx variables accessed when the function is first run), and then is automatically rerun every time any of them change - in this case, it's evaluated any time this.selected changes. The idea of stickySelected was to have a variable that tracks the last thing that was selected, whether or not the selection was cancelled afterwards.

It's quite possible we could do this more elegantly in the future :) Thanks for fixing the crashing when deleting graph - I'm thinking that problem is also related to the fact that we have components that span stores (Meteor and MobX), and I think it's quite possible that we need to do another smaller refactor some time in the future. But I think for now, as long as it works, and let's us rapidly refactor the kind of interfaces we want, it's OK - I will aim for as good code as possible of course, but I will not be too worried if it ends up being a bit ugly or copy and paste for now, as long as user experience goes up...